### PR TITLE
Rename el-4.0 to expressionLanguage-4.0

### LINKED ISSUE #13752: rename el-4.0 feature to expressionLanguage-4.0
Related to discussions:
https://github.com/OpenLiberty/open-liberty/issues/13327
https://github.com/OpenLiberty/open-liberty/issues/13393

We need to make similar changes here that we made for jsp -> pages here: https://github.com/OpenLiberty/open-liberty/issues/13521

I'll be adding https://github.com/OpenLiberty/open-liberty/issues/14318 after this is delivered for the `WLP-AlsoKnownAs: el-4.0`

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #13752

### CODE REVIEW

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#bulid

**Comment by pnicolucci** [comment]
#build

**Review by volosied — APPROVED** [review]
I think you covered everything.
- bnd.bnd (symbolic name & tested features)
- .project files 
- ibm feature names
- file names & folder names
- feature replacement name 

looks good to me, thanks!

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.expressionLanguage-4.0.feature [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.jakarta.expressionLanguage-4.0
+singleton=true
+-bundles=io.openliberty.jakarta.expressionLanguage.4.0; location:="dev/api/spec/,lib/"; mavenCoordinates="org.apache.tomcat:tomcat-el-api:10.0.0-M7"
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.expressionLanguage-4.0.feature:4]: Are those the correct maven co-ords? Isn't it `jakarta.el:jakarta.el-api:4.0.0` ?

**Comment by tevans78** [comment]
What is the overlaid file for `dev/io.openliberty.org.apache.jasper.expressionLanguage.4.0/src/org/apache/el/lang/ELSupport.java`? Is there an apache issue or pull request open to push whatever the changes are back upstream?

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.expressionLanguage-4.0.feature (hunk 1 above)
Comment (pnicolucci) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.expressionLanguage-4.0.feature:4]: These are the correct coordinates, we pull the el api/impl from tomcat
 as they must come from the same source (some impl details in the api for EL).

**Comment by pnicolucci** [comment]
@tevans78  there is an issue opened https://github.com/OpenLiberty/open-liberty/issues/14232 however tomcat rejected our proposal as it was made years ago and we just have not pulled in a newer version. It looks like the last time we updated the EL we had to perform the same overlay. We'll use the issue that is opened to try and see if we can update our JSP impl to not depend on this part of the EL impl. However it may not be possible we need to investigate further. For the time being we have the overlay to move forward.

**Review by tevans78 — APPROVED** [review]

**Review by cbridgha — APPROVED** [review]
LGTM

**Comment by pnicolucci** [comment]
#run-libby-bot

**Pull request merged**